rs/bitcoin/kyt/src/state.rs

```diff
@@ -0,0 +1,179 @@
+use crate::GetTxError;
+use bitcoin::{Address, Transaction};
+use ic_btc_interface::Txid;
+use std::cell::RefCell;
+use std::collections::BTreeMap;
+
+#[derive(Debug, Clone)]
+pub enum FetchTxStatus<T> {
+    PendingOutcall,
+    PendingRetry { buffer_size: u32 },
+    Error(GetTxError),
+    Fetched(FetchedTx<T>),
+}
+
+#[derive(Debug, Clone)]
+pub struct FetchedTx<T> {
+    pub tx: T,
```

Comment:
A mock transaction type makes it easier to write test cases. It is not easy to work with the real `Transaction` type from bitcoin create when coming up with test cases.

Reply:
@gregorydemay Thanks for suggesting to remove the generics. It turns out to be not as difficult as I initially thought, because the computation & verification of txid is actually abstracted out in the get_tx function only. So tests have a lot more flexibility in accepting mocked transactions. Removing the generics is a nice simplification!

Reply:
Thanks a lot for the change, it does look much simpler now!